From 230f2ee5cdec1e2c2069186d1ecfc89c9f5a761f Mon Sep 17 00:00:00 2001
From: Jun Nie <jun.nie@linaro.org>
Date: Fri, 12 Jul 2019 12:19:26 +0800
Subject: [PATCH] generic_boot: init new dtb if CFG_DT_ADDR defined

Init new dtb if CFG_DT_ADDR defined because dtb data from previous
boot may not be cleared yet after reboot. DT overlay may be appended
to dtb at this address many times in reboot test.

For append DT overlay to dtb from previous stage boot loader, arg2
of optee is supposed to be the parameter of dtb address, not CFG_DT_ADDR.

Signed-off-by: Jun Nie <jun.nie@linaro.org>
---
 core/arch/arm/kernel/generic_boot.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/kernel/generic_boot.c b/core/arch/arm/kernel/generic_boot.c
index 2757b1f..fbd8166 100644
--- a/core/arch/arm/kernel/generic_boot.c
+++ b/core/arch/arm/kernel/generic_boot.c
@@ -564,12 +564,14 @@ static int init_dt_overlay(struct dt_descriptor *dt, int __maybe_unused dt_size)
 	int fragment;
 	int ret;
 
+#ifndef CFG_DT_ADDR
 	ret = fdt_check_header(dt->blob);
 	if (!ret) {
 		fdt_for_each_subnode(fragment, dt->blob, 0)
 			dt->frag_id += 1;
 		return ret;
 	}
+#endif
 
 #ifdef CFG_DT_ADDR
 	return fdt_create_empty_tree(dt->blob, dt_size);
-- 
2.7.4

